sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -105,6 +107,46 @@ def _create_open_search_client():
     )
 
 
+def initialize_data_processing_status(filename: str):
+    """Generate data that will be sent to database.
+
+    Parameters
+    ----------
+    filename : str
+        filename of injested data.
+
+    Returns
+    -------
+    dict
+        data for database
+    """
+    # Get instrument name by parsing the filename.
+    # Eg. filename = imap_l0_sci_codice_20230602_v02.pkts
+    # Get instrument name, codice, from the filename.
+    instrument_name = filename.split("_")[3]
```

Comment:
Maybe pass in the `metadata` variable gathered from `_check_for_matching_filetype` function.  Then you can get the level/instrument via `metadata['instrument']` and `metadata['level']`.  That would be useful if the file format ever changed (or for future, non-IMAP missions)

Reply:
That's a good idea! It looks like we get metadata in above lines. I will use that `metadata` and pass that as input to this function.